demisto_sdk/commands/validate/initializer.py

```diff
@@ -252,30 +252,36 @@ def get_unfiltered_changed_files_from_git(self) -> Tuple[Set, Set, Set]:
             debug=True,
             get_only_current_file_names=False,
         )
+
+        """
+        If this command runs on a build triggered by an external contribution PR,
+        the relevant modified files may have an "untracked" status in git.
+        The following code segment retrieves all relevant untracked files that were changed in the
+        external contribution PR. See CIAC-10968 for more info.
+
+        This code snippet ensures that the number of fetched files matches the number of files in
+        the contribution_files_relative_paths.txt file. If a discrepancy is found, indicating untracked files,
+        it raises a ValueError to halt execution due to a mismatch in file counts.
+        See CIAC-12482 for more info.
+        """
         if os.getenv("CONTRIB_BRANCH"):
-            """
-            If this command runs on a build triggered by an external contribution PR,
-            the relevant modified files would have an "untracked" status in git.
-            The following code segment retrieves all relevant untracked files that were changed in the external contribution PR
-            See CIAC-10968 for more info.
-            """
-            logger.info(
-                "\n<cyan>CONTRIB_BRANCH environment variable found, running validate in contribution flow "
-                "on files staged by Utils/update_contribution_pack_in_base_branch.py (Infra repository)</cyan>"
-            )
-            # Open contribution_files_paths.txt created in Utils/update_contribution_pack_in_base_branch.py (Infra) and read file paths
-            relative_untracked_files_paths: Set[Path] = set()
             with open(
                 "contribution_files_relative_paths.txt", "r"
             ) as contribution_file:
-                for single_line in contribution_file:
-                    clean_line: str = single_line.rstrip("\n")
-                    relative_untracked_files_paths.add(Path(clean_line))
-            logger.info(
-                f"\n######## - Added untracked:\n{relative_untracked_files_paths}"
-            )
-            # modified_files = modified_files.union(relative_untracked_files_paths)
-            added_files = set(added_files).union(relative_untracked_files_paths)
+                contribution_files_relative_paths_count_lines = sum(
+                    1 for line in contribution_file if line.strip()
+                )
+
+            if contribution_files_relative_paths_count_lines != (
+                len(modified_files) + len(added_files) + len(renamed_files)
+            ):
```

Comment:
Is there a possibility that the same file could be in more than one set?
if yes, is could mess up our file count, if no, then we are good.
If this case is possible, maybe we could unify all three sets to eliminate dupes and then measure the length.